packages/eslint-plugin-formatjs/rules/enforce-id.ts

```diff
@@ -106,8 +134,22 @@ export default {
     ],
   },
   create(context) {
+    const tmp = context?.options?.[0]
+    const opts = {
+      idInterpolationPattern: tmp?.idInterpolationPattern,
+    } as Opts
+    if (Array.isArray(tmp?.idWhitelist)) {
+      const {idWhitelist} = tmp
+      opts.idWhitelistRegexps = idWhitelist.map(
+        (str: string) => new RegExp(str, 'i')
+      )
+      opts.idWhitelistErrorMsg = ` or does not match whitelisted patterns ["${idWhitelist.join(
```

Comment:
can you just collapse this into the message above? why does this need to be passed in?

Reply:
also do you mean `AND does not match whitelisted patterns...`?

Reply:
> can you just collapse this into the message above? why does this need to be passed in?

It was done for performance reasons. But yep, it makes code less readable. Refactored! Please check a new version.

> also do you mean AND does not match whitelisted patterns...?

OR/AND is quite ambiguous in meanings. I rephrased the message a little bit. Users will receive the following error:
```
"id" does not match with hash pattern [sha512:contenthash:base64:6] or whitelisted patterns ["/\./i", "/^payment_.*/i"].
     Expected: j9qhn+
     Actual: undefined
```
If you have ideas on how to make it more clear, I'm ready to change it ;)